### ISSUE TITLE
CCL_ERROR_SPLINE in CalculatorCosmology when passing decreasing a

### ISSUE THREAD

**Issue by tilmantroester** [LSSTDESC/CCL#863]
Passing the `a` array in `pk_nonlin` in decreasing order causes a CCL_ERROR_SPLINE:

```python
decreasing_a = np.linspace(0.1, 1.0, 100)[::-1]
k = np.logspace(-3, 1, 50)
dummy_Pk = np.ones((len(increasing_a), len(k)))

ccl_cosmo = ccl.CosmologyCalculator(Omega_c=0.25,
                                    Omega_b=0.05,
                                    A_s=2.1e-9,
                                    n_s=0.97,
                                    h=0.7,
                                    pk_nonlin={"a" : decreasing_a, 
                                               "k" : k, 
                                               "delta_matter:delta_matter": dummy_Pk})
```

Error:
```
---------------------------------------------------------------------------
CCLError                                  Traceback (most recent call last)
<ipython-input-51-c6c6bf0dac6e> in <module>
     10                                     pk_nonlin={"a" : decreasing_a, 
     11                                                "k" : k,
---> 12                                                "delta_matter:delta_matter": dummy_Pk})

~/Codes/miniconda/envs/analysis/lib/python3.7/site-packages/pyccl/core.py in __init__(self, Omega_c, Omega_b, h, n_s, sigma8, A_s, Omega_k, Omega_g, Neff, m_nu, m_nu_type, w0, wa, T_CMB, background, growth, pk_linear, pk_nonlin, nonlinear_model)
   1152             self._init_pklin(pk_linear)
   1153         if has_pknl:
-> 1154             self._init_pknl(pk_nonlin, has_nonlin_model)
   1155         self._apply_nonlinear_model(nonlinear_model)
   1156 

~/Codes/miniconda/envs/analysis/lib/python3.7/site-packages/pyccl/core.py in _init_pknl(self, pk_nonlin, has_nonlin_model)
   1285                       a_arr=pk_nonlin['a'], lk_arr=lk, pk_arr=pk,
   1286                       is_logp=use_log, extrap_order_lok=1,
-> 1287                       extrap_order_hik=2, cosmo=None)
   1288             self._pk_nl[n] = pk
   1289         # Set non-linear power spectrum as initialized

~/Codes/miniconda/envs/analysis/lib/python3.7/site-packages/pyccl/pk2d.py in __init__(self, pkfunc, a_arr, lk_arr, pk_arr, is_logp, extrap_order_lok, extrap_order_hik, cosmo, empty)
    106                                                         int(extrap_order_hik),
    107                                                         int(is_logp), status)
--> 108         check(status)
    109         self.has_psp = True
    110 

~/Codes/miniconda/envs/analysis/lib/python3.7/site-packages/pyccl/pyutils.py in check(status, cosmo)
     46     # Check for known error status
     47     if status in error_types.keys():
---> 48         raise CCLError("Error %s: %s" % (error_types[status], msg))
     49 
     50     # Check for unknown error

CCLError: Error CCL_ERROR_SPLINE: 
```

**Comment by damonge** [comment]
This is meant to happen, so is your point that we should throw a more informative error message? I would agree.

**Comment by tilmantroester** [comment]
Yes, there is an explicit error message when passing the array in wrong order for `background`. I can have a look at this.

**Comment by tilmantroester** [comment]
Fixed with #864.
